samtranslator/translator/translator.py

```diff
@@ -46,8 +46,10 @@ def translate(self, sam_template, parameter_values):
         :returns: a copy of the template with SAM resources replaced with the corresponding CloudFormation, which may \
                 be dumped into a valid CloudFormation JSON or YAML template
         """
-        parameter_values = self._add_default_parameter_values(sam_template, parameter_values)
-        parameter_values = self._add_pseudo_parameter_values(parameter_values)
+        sam_parameter_values = SamParameterValues(parameter_values)
```

Comment:
Could you instead add 2 boolean parameters to the constructor (add_default_parameters, and add_psuedo_parameters). When these are true, call `self._ add_default_parameter_values` and `self._ add_pseudo_parameter_values` respectively. SAM already has a lot of mutating calls, but I'd prefer to mitigate where we can.

Reply:
I used `copy.deepcopy` so it won't modify the original parameter_values. I actually prefer to leave those two methods separated instead of bundling them together to give the caller more flexibility to decide whether to just use one of them or both of them